src/agni/server/manager/UserManager.java

```diff
@@ -16,8 +16,69 @@ public UserManager(InfoSender infoSender,
     }
 
     @Override
-    public void userRequest(String ip, byte type) {
+    public void userRequest(String ip, byte type, String message) {
         // TODO Auto-generated method stub
+    	String username = userDataGuard.getUsername(ip); 
+        if(type == 0x00) { 			// join chat; message = group to join
+        	String groupName = message; 
+        	if (userDataGuard.groupExists(groupName)) {
+        		userDataGuard.addUserToChat(username,groupName); 
+        		infoSender.sendInfo(ip, "success: joining " + groupName);
+        	}
+        	else {
+        		infoSender.sendInfo(ip, "failed: " + groupName + " doesn't exist");
+        	}
+        }
+        else if (type == 0x01){ 	// leave chat 
+        	String groupName = message; 
+        	if (userDataGuard.groupExists(groupName)) {
+        		userDataGuard.removeUserFromChat(username,groupName); 
+        		infoSender.sendInfo(ip, "success: leaving " + groupName);
+        	}
+        	else {
+        		infoSender.sendInfo(ip, "failed: " + groupName + " doesn't exist");
+        	}
+        	
+        }
+        else if (type == 0x02){ 	// see friends list 
+        	String friendList = userDataGuard.getFriendList(username); 
+        	// TODO get status of each friend
+        	//String statusList = userDataGuard.getStatusList(username); 
+        	infoSender.sendInfo(ip, "friends:" + friendList);
+        }
+        else if (type == 0x03){		// check friend status ; message = friend to check 
+        	String friend = message;
+        	if (userDataGuard.isOnline(friend)) {
+        		infoSender.sendInfo(ip, friend + " online");
+        	}
+        	else {
+        		infoSender.sendInfo(ip, friend + " offline");
+        	}
+        }
+        else if (type == 0x04){		// add friend 
+        	String friend = message;
+        	if (userDataGuard.userExists(message)) {
+        		
+        		String friendList = userDataGuard.getFriendList(username); 
+        		if(friendList.contains(friend)) {
+        			infoSender.sendInfo(ip, friend + " already added");
```

Comment:
Probably send differing messages depending on whether 2 people are already friends or whether their request is pending.